pkg/openstack/instances.go

```diff
@@ -707,5 +749,54 @@ func getAttachedPorts(client *gophercloud.ServiceClient, serverID string) ([]por
 		DeviceOwner: "compute:nova",
 	}
 
-	return openstack.GetPorts(client, listOpts)
+	interfaces, err := openstack.GetPorts(client, listOpts)
+	if err != nil {
+		return interfaces, err
+	}
+	subInterfaces, err := getSubInterfaces(client, interfaces)
+	if err != nil {
+		return interfaces, err
+	}
+	interfaces = append(interfaces, subInterfaces...)
+
+	return interfaces, nil
+}
+
+// getSubInterfaces returns the sub-interfaces attached via trunks to the specified interfaces
+func getSubInterfaces(client *gophercloud.ServiceClient, interfaces []ports.Port) ([]ports.Port, error) {
+	var subports []ports.Port
+	var portExt struct {
+		ports.Port
+		trunk_details.TrunkDetailsExt
+	}
+	for _, iface := range interfaces {
+		// Check if trunk is attached to the interface
+		err := ports.Get(client, iface.ID).ExtractInto(&portExt)
```

Comment:
could we do this trunk call already in row 752? include trunk info in GetPorts? By that way we could have the same info in single api call instead of own for each interface

Reply:
https://github.com/kubernetes/cloud-provider-openstack/blob/master/pkg/util/openstack/network.go#L149

```
	allPorts, err := neutronports.ExtractPortsInto(allPages, structhere)
```

could do that.

Reply:
blah, looks like trunk information is available only in single GET call and not in list call at all... anyway I am not quite huge fan of this implementation. It will add unnecessary call for most of the users. Could we have somekind of way to get rid of this? Like config item in occm to enable trunk calls?

Reply:
Good point, checking and will do.

Reply:
It is returned in list queries, as tested in the gophercloud acceptance test: https://github.com/gophercloud/gophercloud/blob/master/acceptance/openstack/networking/v2/extensions/trunk_details/trunks_test.go

Only the MAC address is missing, and we don't need that here.

Reply:
Apparently, I missed the extension to list, I will check and update.

Reply:
@mdbooth then perhaps something is missing from api docs, this is only which I found https://docs.openstack.org/api-ref/network/v2/index.html?expanded=show-trunk-details-detail and that is for single port. I do not have environment which contains trunk ports, so I cannot verify does it really work in list queries as well.

Reply:
The gophercloud acceptance tests run against an installed OpenStack, so they reflect reality. They are returned for LIST.

Reply:
comment taken.